### PR TITLE
CAE description integration

### PR DESCRIPTION
**Pull request by CMiSSioN**
bonus description integration for CAE

### CODE REVIEW

File: source/Features/OverrideDescriptions/BonusDescriptions.cs [hunk 1]
```diff
@@ -7,127 +7,145 @@
...
+    public Dictionary<UnitType, List<BonusDescriptionContent>> CriticalEffects { get; set; }
+    public string Description { get; private set; }
+    public ExtendedDescription(string original) {
+      CriticalEffects = new Dictionary<UnitType, List<BonusDescriptionContent>>();
+      Traits = new List<BonusDescriptionContent>();
+      Description = original;
+    }
+  }
+  public static class DescriptionsHelper {
+    private static Dictionary<string, ExtendedDescription> extendedDescriptions = new Dictionary<string, ExtendedDescription>();
```
Comment (CptMoore) [inline source/Features/OverrideDescriptions/BonusDescriptions.cs]: Better to move this to CC field, static not needed

File: source/Features/OverrideDescriptions/BonusDescriptions.cs [hunk 2]
```diff
@@ -7,127 +7,145 @@
...
+    public static void AddBonusDescription(this DescriptionDef def, BonusDescriptionType type, UnitType ut, string template, string elements, string originalDescription) {
+      //Control.mod.Logger.LogDebug("AddBonusDescription: "+def.Id+" type:"+type+" unit:"+ut+" template:"+template+" elements:"+elements+" original:"+originalDescription);
+      if (extendedDescriptions.ContainsKey(def.Id) == false) { extendedDescriptions.Add(def.Id, new ExtendedDescription(originalDescription)); };
+      ExtendedDescription descr = extendedDescriptions[def.Id];
+      switch (type) {
+        case BonusDescriptionType.Traits: descr.Traits.Add(new BonusDescriptionContent(template,elements)); break;
+        case BonusDescriptionType.CriticalEffects:
```
Comment (CptMoore) [inline source/Features/OverrideDescriptions/BonusDescriptions.cs]: probably also not needed, better to just have CC fields
comp.Is<BonusDescription>(our var e)
e.CriticalEffect
e.Trait
or e.CriticalEffectsFor(actor)

Comment (CptMoore) [reply source/Features/OverrideDescriptions/BonusDescriptions.cs]: enum wih where to show:
Everywhere
mechlab
combatdesciption
statusdescription

enum actor:
ALL
Mech
Turret
Vehicle

**Comment by CptMoore** [comment]
the solution in CC is good enough right?

**Comment by CMiSSioN** [comment]
i have no issues

**Pull request closed without merging**